hw_7_classes.py

```diff
@@ -0,0 +1,233 @@
+        self._last_ocktets = []
+
+    def get_ip_list(self):
+        return f"IP addresses: {self._list}"
+
+    def get_reversed_ip_addresses(self):
+        for address in self._list:
+            self._reversed_addr.append('.'.join(address.split('.')[::-1]))
+        return f"Reversed addresses list: {self._reversed_addr}"
+
+    def get_last_3_ocktets_list(self):
+        for address in self._list:
+            self._last_3_ocktets.append('.'.join(address.split('.')[1:]))
+        return f"Last 3 ocktets: {self._last_3_ocktets}"
+
+    def get_last_ocktets_list(self):
+        for address in self._list:
+            self._last_ocktets.append(address.split('.')[-1])
+        return f"Last ocktets: {self._last_ocktets}"
+
+
+print('\nTask 1. IP addresses class\n')
+a = IpAddress()
+print(a.get_ip_list())
+print(a.get_reversed_ip_addresses())
+print(a.get_last_3_ocktets_list())
+print(a.get_last_ocktets_list())
+
+
+# Задача-2
+# У вас несколько JSON файлов. В каждом из этих файлов есть
+# произвольная структура данных. Вам необходимо написать
+# класс который будет описывать работу с этими файлами, а
+# именно:
+# 1) Запись в файл
+# 2) Чтение из файла
+# 3) Объединение данных из файлов в новый файл
+# 4) Получить путь относительный путь к файлу
+# 5) Получить абсолютный путь к файлу
+
+class JsonHandler:
+
+    default_file = 'example_json_1.json'
+
+    def __init__(self, file):
+        if file:
+            self._file = file
+        else:
+            self._file = JsonHandler.default_file
+
+    @staticmethod
+    def find_path_to_file(file_name, fpath=os.getcwd()):
+        """Function accepts file name and path to start search from with default value set to current directory.
+        Examples: 'D:' , 'D:\folder'
+        Returns absolute path to file or raises FileNotFoundError
+        """
+        print(f'\nSearching: {file_name}...')
+        fpath.replace('\\', '\\\\')
+        glob_path = pathlib.Path(fpath)
+        file_path = None
+        for path in sorted(glob_path.rglob('*.json')):
+            if str(file_name) in str(path):
+                # print(f'Requested file is located at: {str(path)}\n')
+                file_path = str(path)
+        if not file_path:
+            print(f'File {file_name} was not found in {fpath}')
+            raise FileNotFoundError
+        return file_path
+
+    def read_json(self, file):
+        my_printer = pprint.PrettyPrinter(width=90, sort_dicts=False)
+        try:
+            with open(self.find_path_to_file(file)) as f:
+                data = json.load(f)
+                my_printer.pprint(data)
+        except IOError:
+            print(f'File {file} not found')
+
+    def write_json(self, fdata, trg):
+        if os.path.isfile(trg):
+            print('\nFile exists. Overwriting...')
+        else:
+            print("File doesn't exist. Creating a new file")
+        with open(trg, 'w') as output_file:
+            json.dump(fdata, output_file)
+
```

Comment:
It would be nice to add one more method for JSON validation to catch JSONDecodeError